### PR TITLE
feat(cx): implement fdx-side-nav component

### PR DESCRIPTION
**Pull request by mikerodonnell89**
fixes none

New implementation of the side nav. You will see a good deal of duplicate code from core's side nav. But as this will replace core's I think it is fine for now rather than abstracting the shared behavior out, only to deprecate core's in the future.

### CODE REVIEW

File: libs/cx/src/lib/nested-list/nested-link/nested-link.component.ts [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+        // eslint-disable-next-line @angular-eslint/component-selector
+        '[fdx-nested-list-link], button[fdx-nested-list-link], a[fdx-nested-list-link], div[fdx-nested-list-link]',
+    host: {
+        '[attr.tabindex]': '!_nestedListButton ? 0 : -1'
+    }
+})
+export class NestedLinkComponent {
+    /** Function that is called on click event dispatch on this element. */
+    @Input()
+    onClickCallback: () => void;
```
Comment (platon-rov) [inline libs/cx/src/lib/nested-list/nested-link/nested-link.component.ts:43]: Why this is not `EventEmitter`? Accepting callback is fine, though it's kinda different principle than emitting events.

File: libs/cx/src/lib/nested-list/nested-list/nested-list.component.ts [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+import { FdLanguage, FD_LANGUAGE, TranslationResolver } from '@fundamental-ngx/i18n';
+import {
+    ContentDensityObserver,
+    contentDensityObserverProviders,
+    ContentDensityMode
+} from '@fundamental-ngx/core/content-density';
+
+@Component({
+    template: `
+        <!--li fdx-nested-list-item *ngIf="popoverTitle" class="fdx-nested-list__item--header">
```
Comment (platon-rov) [inline libs/cx/src/lib/nested-list/nested-list/nested-list.component.ts:34]: Maybe we can remove commented stuff?

File: libs/cx/src/lib/nested-list/prepared-nested-list/prepared-nested-list.component.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+import { NestedListComponent } from '../nested-list/nested-list.component';
+import { NestedListItem, NestedListModel } from '../nested-list-model';
+
+/**
+ * Component for internal usage, allows to generate the nested list from defined object.
+ */
+@Component({
+    selector: 'fdx-prepared-nested-list',
+    templateUrl: './prepared-nested-list.component.html',
+    styleUrls: ['./prepared-nested-list.component.scss'],
```
Comment (platon-rov) [inline libs/cx/src/lib/nested-list/prepared-nested-list/prepared-nested-list.component.ts:11]: It's empty, can we remove it?

**Review by platon-rov — COMMENTED** [review]
Good job!

Please fix failing pipeline

File: versions.json [hunk 4]
```diff
@@ -1,114 +1,114 @@
...
-    "id": "0.12.0",
-    "url": "https://5db1dc978d2a340009a82d64--fundamental-ngx.netlify.app/"
-  },
-  {
-    "id": "0.11.0",
-    "url": "https://5d8a3409acaf8d00070ccd64--fundamental-ngx.netlify.app/"
-  }
-]
\ No newline at end of file
+    {
```
Comment (droshev) [inline versions.json:2]: why is this file changed

Comment (mikerodonnell89) [reply versions.json:2]: Formatting maybe from linting or migrations

Comment (droshev) [reply versions.json:2]: Can you revert it?

**Review by droshev — CHANGES_REQUESTED** [review]
- The PR(and the message during merge) needs scope. For Example `feat(cx): introduce side navigation component`

Few visuals issues:

- hovering the arrow makes the button look weird 
Screenshot 2022-12-21 at 9 10 02 PM

- on scrolling up we we see the icons behind the up/down button
Screenshot 2022-12-21 at 9 12 31 PM
side-nav-icon-behind

 
- weird focus 
Screenshot 2022-12-21 at 9 13 15 PM
Screenshot 2022-12-21 at 9 18 46 PM


- the arrow button jumps up/down when we move up/down  the last icon
side-nav-arrow-jumps

File: versions.json (hunk 4 above)
Comment (mikerodonnell89) [reply versions.json:2]: Not without bypassing the linting tools we have in place

**Comment by mikerodonnell89** [comment]
> * hovering the arrow makes the button look weird
>   Screenshot 2022-12-21 at 9 10 02 PM
> * on scrolling up we we see the icons behind the up/down button
>   Screenshot 2022-12-21 at 9 12 31 PM
>   side-nav-icon-behind
>         
>           [
>             
>               side-nav-icon-behind
>             
>           ](https://user-images.githubusercontent.com/4380815/209040661-350335bc-610d-40c2-b468-183951bbec76.gif)
>           
>           
>             
>               
>                 
>               
>               
>                 
>                 
>               
>             
>             [
>               
>                 
>               
>             ](https://user-images.githubusercontent.com/4380815/209040661-350335bc-610d-40c2-b468-183951bbec76.gif)
> * weird focus
>   Screenshot 2022-12-21 at 9 13 15 PM
>   Screenshot 2022-12-21 at 9 18 46 PM

These are all essentially the same issue, and will require refactors from fundamental-styles. The navigable items need their own container separate from the up/down buttons

**Comment by g-cheishvili** [comment]
Check "Narrow" example with keyboard:
- Focus any item
- Move down to the item which is expandable
- Click enter on space - no effect
- Click right arrow - it opens
- Now try to navigate to the items inside that expanded list < -- This is a problem

File: libs/cx/src/lib/side-navigation/side-navigation.component.ts [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+        if (this.sideNavigationConfiguration) {
+            this.keyboardService.refreshItems(this.getLists());
+        }
+    }
+
+    /** @hidden */
+    @HostListener('window:resize')
+    onResize(): void {
+        if (this.collapseWidth) {
+            this.condensed = window.innerWidth <= this.collapseWidth;
```
Comment (g-cheishvili) [inline libs/cx/src/lib/side-navigation/side-navigation.component.ts:123]: Do not call `window` directly. Use abstraction of Window

File: libs/cx/src/lib/side-navigation/side-navigation.component.ts [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+    @ContentChild(SideNavigationMainComponent, { descendants: true })
+    sideNavMain: SideNavigationMainComponent;
+
+    /** @hidden */
+    @ViewChildren(PreparedNestedListComponent)
+    preparedNestedList: QueryList<PreparedNestedListComponent>;
+
+    /** @hidden */
+    constructor(private keyboardService: NestedListKeyboardService, private nestedListState: NestedListStateService) {
+        this.keyboardService.refresh$.subscribe(() => {
```
Comment (g-cheishvili) [inline libs/cx/src/lib/side-navigation/side-navigation.component.ts:82]: Finalize this somehow. Either with KeyboardService's ngOnDestroy complete, either unsubscribe from this class

File: libs/cx/src/lib/side-navigation/side-navigation.component.ts (hunk 5 above)
Comment (mikerodonnell89) [reply libs/cx/src/lib/side-navigation/side-navigation.component.ts:133]: We will address this app-wide in another PR

**Comment by InnaAtanasova** [comment]
One minor issue from my side, the rest looks great!

There are list elements under the overflow buttons and when they are focussed, the outline looks broken. In my opinion the area that contains the side nav items should be between the buttons. Currently the scroll area includes the buttons, which causes the issue.
Screenshot 2023-01-06 at 12 27 56 PM
Screenshot 2023-01-06 at 12 28 46 PM

**Review by InnaAtanasova — APPROVED** [review]

**Review by droshev — APPROVED** [review]
